src/main/kotlin/app/revanced/patches/youtube/ad/infocards/patch/HideInfocardsPatch.kt

```diff
@@ -0,0 +1,74 @@
+package app.revanced.patches.youtube.ad.infocards.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.extensions.instruction
+import app.revanced.patcher.extensions.replaceInstruction
+import app.revanced.patcher.fingerprint.method.impl.MethodFingerprint.Companion.resolve
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultError
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.util.smali.ExternalLabel
+import app.revanced.patches.youtube.ad.infocards.annotations.HideInfocardsCompatibility
+import app.revanced.patches.youtube.ad.infocards.fingerprints.InfocardsIncognitoFingerprint
+import app.revanced.patches.youtube.ad.infocards.fingerprints.InfocardsMethodCallFingerprint
+import app.revanced.patches.youtube.ad.infocards.fingerprints.InfocardsIncognitoParentFingerprint
+import app.revanced.patches.youtube.ad.infocards.resource.patch.HideInfocardsResourcePatch
+import app.revanced.patches.youtube.misc.integrations.patch.IntegrationsPatch
+import org.jf.dexlib2.builder.instruction.BuilderInstruction35c
+
+@Patch
+@DependsOn([IntegrationsPatch::class, HideInfocardsResourcePatch::class])
+@Name("hide-infocards")
+@Description("Hides infocards in videos.")
+@HideInfocardsCompatibility
+@Version("0.0.1")
+class HideInfocardsPatch : BytecodePatch(
+    listOf(
+        InfocardsIncognitoParentFingerprint,
+        InfocardsMethodCallFingerprint,
+    )
+) {
+    override fun execute(context: BytecodeContext): PatchResult {
+        val parentResult = InfocardsIncognitoParentFingerprint.result
+            ?: return PatchResultError("Parent fingerprint not resolved!")
+
+
+        InfocardsIncognitoFingerprint.resolve(context, parentResult.classDef)
+        val result = InfocardsIncognitoFingerprint.result
+            ?: return PatchResultError("Required parent method could not be found.")
+
+        val method = result.mutableMethod
+        val implementation = method.implementation
+            ?: return PatchResultError("Implementation not found.")
```

Comment:
This null check is redundant because the fingerprint contains a string reference which is implicating an implementation. Assert non null here.